test/api/union/user_bin_sequence_test.cpp

```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
     EXPECT_RANGE_EQ(this->get_filenames(), (std::vector<std::string>{"small2.fa", "small2.fa", "small.fa", "small.fa"}));
     EXPECT_RANGE_EQ(this->get_user_bin_kmer_counts(), (std::vector<size_t>{800, 700, 600, 500}));
 }
+
+TEST_F(user_bin_sequence_test, read_hll_files)
+{
+    size_t const k = 16;
+    size_t const b = 4;
+    hyperloglog expected(b);
+
+    std::string input_file = DATADIR"small.fa";
+    sequence_file_type seq_file{input_file};
+
+    // put every sequence in this file into the sketch
+    for (auto && [seq] : seq_file)
+    {
+        // we have to go C-style here for the HyperLogLog Interface
+        char* c_seq_it = &*seq.begin();
+        char* end = c_seq_it + seq.size();
+
+        while(c_seq_it + k <= end)
+        {
+            expected.add(c_seq_it, k);
+            ++c_seq_it;
+        }
+    }
+
+    this->read_hll_files(DATADIR);
+
+    EXPECT_EQ(this->get_sketches().size(), 4u);
+    for (auto const & sketch : this->get_sketches())
+        EXPECT_EQ(sketch.estimate(), expected.estimate());
+}
+
+TEST_F(user_bin_sequence_test, read_hll_files_empty_dir)
+{
+    seqan3::test::tmp_filename tmp_file{"some_file"};
+    auto parent = tmp_file.get_path().parent_path();
+    auto empty_dir = parent/"empty_dir";
+    std::filesystem::create_directory(empty_dir); // create empty dir
```

Comment:
Can you check (on your machine, not as test) that the directory is deleted after execution?
(not sure if `tmp_filename` deletes the whole directory)

Reply:
yes I checked, nothing is remaining in my tmp dir